### PR TITLE
Support schema management

### PR DESCRIPTION
**Pull request by Linary**

### CODE REVIEW

File: hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+        Ex.check(connId != reusedConnId, "schema.conn.cannot-reuse-self");
+        Ex.check(!CollectionUtils.isEmpty(names),
+                 "common.param.cannot-be-empty", "names");
+        return this.elService.checkConflicts(names, reusedConnId, connId);
+    }
+
+    @PostMapping("check_conflict")
+    public ConflictStatus checkConflict(@RequestBody EdgeLabelEntity entity,
+                                        @RequestParam("conn_id") int connId) {
+//        this.checkParamsValid(entity, connId, false);
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java]: uunsed?

File: hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        return this.elService.checkConflict(entity, connId);
+    }
+
+    @PostMapping("reuse")
+    public void reuse(@RequestBody ConflictDetail detail,
+                      @RequestParam("conn_id") int connId) {
+//        Ex.check(!CollectionUtils.isEmpty(names),
+//                 "common.param.cannot-be-empty", "names");
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java]: ditto

File: hubble-be/src/main/resources/i18n/messages_zh_CN.properties [hunk 3]
```diff
@@ -21,15 +25,54 @@ graph-connection.exist.graph-host-port=已存在相同的图名 {0}, 主机名 {
...
+schema.propertykey.not-exist=不存在名字为 {0} 的属性类型
+schema.propertykey.unmatch-regex=属性类型名 {0} 不合法，名字应该由字母，数字、中文、下划线组成，最多128个字符
+schema.propertykey.in-using=属性类型 {0} 正在被使用
+schema.propertykey.reuse.failed=重用属性类型失败
+
+schema.vertexlabel.exist=已存在名字为 {0} 的顶点类型
+schema.vertexlabel.not-exist=不存在名字为 {0} 的顶点类型
+schema.vertexlabel.unmatch-regex=顶点类型名 {0} 不合法，名字应该由字母，数字、中文、下划线组成，最多128个字符
+schema.vertexlabel.in-using=顶点类型 {0} 正在被使用
+schema.vertexlabel.property.cannot-be-null-and-empty=顶点类型 {0} 的属性不能为 null 或空
```
Comment (javeme) [inline hubble-be/src/main/resources/i18n/messages_zh_CN.properties]: 统一为“不能为空”吧

File: hubble-be/src/main/java/com/baidu/hugegraph/config/TomcatServletConfig.java [hunk 4]
```diff
@@ -48,7 +48,8 @@ public void customize(TomcatServletWebServerFactory factory) {
         try {
             factory.setAddress(InetAddress.getByName(host));
         } catch (UnknownHostException e) {
-            throw new ExternalException(String.format("Unknown host %s", host));
+            throw new ExternalException(String.format("Unknown host %s", host),
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/config/TomcatServletConfig.java]: "service.unknown-host"

File: hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/PropertyKeyController.java [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        Ex.check(CollectionUtils.isEmpty(entity.getElEntities()),
+                 "common.param.must-be-null", "edgelabels");
+        return this.service.checkConflict(entity, connId, true);
+    }
+
+    @PostMapping("reuse")
+    public void reuse(@RequestBody ConflictDetail detail,
+                      @RequestParam("conn_id") int connId) {
+//        Ex.check(!CollectionUtils.isEmpty(detail.getPropertyKeyConflicts()),
+//                 "common.param.cannot-be-empty", "names");
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/PropertyKeyController.java]: unused?

File: hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+import com.baomidou.mybatisplus.core.metadata.IPage;
+import com.google.common.collect.ImmutableList;
+
+@RestController
+@RequestMapping(Constant.API_VERSION + "schema/edgelabels")
+public class EdgeLabelController extends SchemaController {
+
+    private static final List<String> PRESET_COLORS = ImmutableList.of(
+            //            "#fa2f4c", "#ff7438", "#ffc900", "#00c22a", "#00bbff",
+            //            "#0079ff", "#5957e0", "#bb4be5", "#ff1c5f"
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java]: unused?

File: hubble-be/src/main/java/com/baidu/hugegraph/license/LicenseVerifyManager.java [hunk 7]
```diff
@@ -128,18 +123,19 @@ private void checkIpAndMac(ExtraParam param) {
         if (StringUtils.isEmpty(expectIp)) {
             return;
         }
+
+        ServerInfo serverInfo = ServerInfo.instance();
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/license/LicenseVerifyManager.java]: don't fetch from static instance

File: hubble-be/src/main/java/com/baidu/hugegraph/entity/schema/Stylefiable.java [hunk 8]
```diff
@@ -0,0 +1,10 @@
+/*
```
Comment (zhoney) [inline hubble-be/src/main/java/com/baidu/hugegraph/entity/schema/Stylefiable.java]: license

File: hubble-be/src/main/java/com/baidu/hugegraph/entity/schema/Timefiable.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (C) 2019 Baidu, Inc. All Rights Reserved.
+ */
+
```
Comment (zhoney) [inline hubble-be/src/main/java/com/baidu/hugegraph/entity/schema/Timefiable.java]: ditto

File: hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/VertexLabelController.java [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+                 "common.param.cannot-be-empty", "vertexlabels");
+        Ex.check(CollectionUtils.isEmpty(entity.getElEntities()),
+                 "common.param.must-be-null", "edgelabels");
+        return this.vlService.checkConflict(entity, connId, true);
+    }
+
+    @PostMapping("reuse")
+    public void reuse(@RequestBody ConflictDetail detail,
+                      @RequestParam("conn_id") int connId) {
+//        Ex.check(!CollectionUtils.isEmpty(names),
```
Comment (zhoney) [inline hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/VertexLabelController.java]: delete if no use

**Pull request merged**